src/main/scala/strawman/collection/BuildFrom.scala

```diff
@@ -0,0 +1,88 @@
+package strawman.collection
+
+import strawman.collection.mutable.Builder
+
+import scala.{Any, Ordering}
+
+/**
+  * Auxiliary type allowing us to define which collection type to build according to the type of an existing collection.
+  *
+  * @tparam From Existing collection type that will drives the collection type to build
+  * @tparam A Element type
+  */
+trait BuildFrom[-From, -A] {
+
+  /** Collection type to build */
+  type To
+
+  def newBuilder(): Builder[A, To]
+
+}
+
+trait BuildFromLowPriority {
+
+  /**
+    * Extracts the binary type constructor of `From` and applies it to `A` and `B`
+    * to build the `To` type.
+    */
+  implicit def binaryTC[CC[_, _], A, B](implicit
+    cb: CanBuild[(A, B), CC[A, B]]
+  ): BuildFrom.Aux[CC[_, _], (A, B), CC[A, B]] =
+    new BuildFrom[CC[_, _], (A, B)] {
+      type To = CC[A, B]
+      def newBuilder(): Builder[(A, B), CC[A, B]] = cb.newBuilder()
+    }
+
+}
+
+object BuildFrom extends BuildFromLowPriority {
+
+  type Aux[From, A, To0] = BuildFrom[From, A] { type To = To0 }
```

Comment:
I think `Build[From, A, To]` ends up seeming too complex even though it is "all parameters".  If the `Aux`-style transformation of a member into a parameter is something that can be behind the scenes in almost every case, I think that would give a better user experience.  If people get to interact with `BuildFrom[From, A]` instead of a triply parameterized thing, I think they'll be happier.  Also, the triply parameterized one is kind of misleading because we don't fill in most of `Build[From, A, To]` in an interesting way for specific choices of `From` and `To`.  Generally if we know what we want to go `To` we don't care where we came from; and if we know where we come `From` there is only a single choice of where to go `To`.